### DISCUSSION TITLE
Using logAttribute name as alias?

### DISCUSSION THREAD

**Discussion by lovecoding-git** [spatie/laravel-activitylog#710] [Q&A] [answered]
`    protected static $logAttributes = ['name', 'type', 'status'];`

Is this possible? or is there any alternative way for it?

`protected static $logAttributes = ['name as other_name', 'type as other_type', 'status as blog_status'];`

**Comment by blindsz** [comment]
You can actually do it inside tapActivity() but having something like that in your example above would be much cleaner.

**Comment by Gummibeer** [comment, marked as answer]
Hey,
that's not possible - except the mentioned `tapActivity()` method and map the array by your own.
Right now I would also say that this is a feature that won't happen - because of a massive increase in complexity without a common usecase.

Could you explain why you would like to alias your model attributes in the log? I can't imagine any case why the log should have aliased attributes. 🤔

**Comment by blindsz** [comment]
> Hey,
> that's not possible - except the mentioned `tapActivity()` method and map the array by your own.
> Right now I would also say that this is a feature that won't happen - because of a massive increase in complexity without a common usecase.
> 
> Could you explain why you would like to alias your model attributes in the log? I can't imagine any case why the log should have aliased attributes. 🤔

Hi. I'm thinking that he may want to rename related attributes for example instead of **client.name: Anderson M. Anderson** he may just want it to be **client: Anderson M. Anderson**

**Comment by lovecoding-git** [comment]
Hi, 
I am going to show this to clients.
I assume clients don't know db column names.
If I show him "Blog Status" instead of "is_approved" or something like that, it would be great.

**Comment by Gummibeer** [comment]
In this case I would do this mapping where it belongs to - in the frontend. For example via translations to support multiple languages.
At the end you should already have this mapping somehow if show forms with labels.

**Comment by blindsz** [comment]
> Hi,
> I am going to show this to clients.
> I assume clients don't know db column names.
> If I show him "Blog Status" instead of "is_approved" or something like that, it would be great.

Save using db fields and map the data when the user wants to see the changes.

**Comment by lovecoding-git** [comment]
> In this case I would do this mapping where it belongs to - in the frontend. For example via translations to support multiple languages.
> At the end you should already have this mapping somehow if show forms with labels.

Thanks
